trace_context/HTTP_HEADER_FORMAT.md

```diff
@@ -144,12 +149,37 @@ boolean traced = (traceFlags & FLAG_TRACED) == FLAG_TRACED
 
 #### Requested Flag (00000001)
 
-When set, the least significant bit recommends the request should be traced. A caller who
-defers a tracing decision leaves this flag unset.
+When set, the least significant bit recommends the request should be traced.
+Typical use for this flag is propagating a sampling decision from the service
+originating the distributed trace. For instance, if originating service has a
+logic to record information about 1 of every 100 requests - the one that was
+marked for recording will have this flag set to `1`.
+
+Many distributed tracing scenarios may be broken when only small subset of calls
+participated in this distributed trace were collected. Thus it is important to
+follow the header mutation rules. This flag can only be changed from `0` to `1`,
+and not the other way around. So the decision of one component to record will be
+respected by every component.
+
+Abuse of this flag - always setting it to `1` will - also lead to broken
+scenarios. Most platforms will respect this flag, but have a protection in place
+to limit the amount of collected traces. And the logic of this limiting may
+differ from component to component. Thus on high volume of traces when different
+components will apply different policies - most traces will not have full information.
+
+Caller who defers a tracing decision SHOULD leaves this flag unset. Caller may
+communicate the priority of deferred decision (most likely will be collected)
+using `recorded` flag.
+
+Flag `recorded` can also be used to communicate the ask to collect information
+or notify that information was collected and available for querying from the
+caller service.
 
 #### Recorded Flag (00000010)
```

Comment:
I think this flag should be removed as it adds little value and results in a hex character "03" which is not intuitive either

Reply:
The problem this flag addresses is the ability for vendor to communicate priority of deferred decision. With the semantic "if in doubt - try to collect". See this:

> Caller who defers a tracing decision SHOULD leaves this flag unset. Caller may
communicate the priority of deferred decision (most likely will be collected)
using `recorded` flag.
> Flag `recorded` can also be used to communicate the ask to collect information
or notify that information was collected and available for querying from the
caller service. 

Basically it is a relieve valve to minimize the abuse of `requested` flag.

I understand the problem of `03`, but if we treat those two characters as bitmask - there is very little we can do with this problem unless we are sure there will be no more flags. In fact having two flags not will find out "bad implementations" earlier.